src/MacroTools/Mechanics/DemonGates/DemonGateBuff.cs

```diff
@@ -85,10 +85,11 @@ public DemonGateBuff(unit caster, int demonUnitTypeId, float spawnInterval, int
     public override void OnApply()
     {
       Target
-        .IssueOrder("setrally", Target.GetPosition())
+        .IssueOrder(852002, Target.GetPosition())
         .SetMaximumMana((int)_spawnInterval)
         .AddAbility(_toggleAbilityTypeId)
         .IssueOrder("immolation");
+      Progress = _spawnInterval / 2;
```

Comment:
I think instead of this workaround for mana getting wiped out, you could just fix it. With your fix to the `OnStack` method, you could just have it so that the new buff gets it Progress set to the value of the old buff, thus preserving Progress.

Reply:
I tried that but I couldn't seem to get it to work, presumably because the new Buff has new progress.

Reply:
Yeah, tried having another go at it but would need some help to look at it which I don't think either of us want to put in. I think this is an acceptable solution in terms of gameplay. Although having it reset to 0 wouldn't be game breaking either, having something here _might_ remind us to look into it later.